Jenkinsfile

```diff
@@ -1,170 +1,163 @@
+							sh "sudo bash ${env.JENKINS_HOME_PATH}/changeGraphDatabase.sh"
 							sh "sudo service neo4j start"
 							sh "sudo service tomcat7 start"
-							sh "rm ${env.JENKINS_HOME_PATH}changeGraphDatabase.sh"
+							sh "rm ${env.JENKINS_HOME_PATH}/changeGraphDatabase.sh"
 						}
 					}
 				}
 			}			
 		}
-		stage('Setup: Generate Analysis.bin file'){
+		// Generates an intermediate analysis.bin file that is used by the biomodels-mapper.
+		stage('Setup: Generate BioModels analysis.bin file'){
 			steps{
 				script{
-					cloneOrPullGitRepo("analysis-core")
+					def releaseVersion = utils.getReleaseVersion()
+					utils.cloneOrUpdateLocalRepo("analysis-core")
+					// Builds analysis-core jar file and runs program that builds the analysis-biomodels-vXX.bin file.
 					dir("analysis-core"){
-						def analysisBinName = "analysis-biomodels-v${currentRelease}.bin"
-						sh "mvn clean compile assembly:single"
+						def analysisBinName = "analysis-biomodels-v${releaseVersion}.bin"
+						// Builds jar file for analysis-core.
+						utils.buildJarFile()
+						
 						withCredentials([usernamePassword(credentialsId: 'neo4jUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
 							sh "java -jar target/analysis-core-jar-with-dependencies.jar --user $user --password $pass --output ./${analysisBinName} --verbose"
 						}
+						// Symlinks the analysis-biomodels-vXX.bin file to the generic analysis.bin path.
 						sh "cp ${analysisBinName} ${env.ANALYSIS_SERVICE_INPUT_ABS_PATH}/"
 						sh "ln -sf ${env.ANALYSIS_SERVICE_INPUT_ABS_PATH}/${analysisBinName} ${env.ANALYSIS_SERVICE_INPUT_ABS_PATH}/analysis.bin"
+						// Restart tomcat7 so that updated graph DB and analysis bin is being used.
 						sh "sudo service tomcat7 stop"
 						sh "sudo service tomcat7 start"
 					}
 				}
 			}
 		}
-		stage('Setup: Build jar file'){
-			steps{
-				script{
-          				sh 'mvn clean compile assembly:single'
-				}
-			}
-		}
+		// Runs the perl 'biomodels.pl' script that builds the models2pathways.tsv file from release_current DB.
 		stage('Main: Generate BioModels file'){
 			steps{
 				script{
+					def releaseVersion = utils.getReleaseVersion()
+					
 					dir("${env.ABS_RELEASE_PATH}/biomodels/"){
+						// Downloads the BioModels_Database-r31_pub-sbml_files that are stored on S3.
+						def biomodelsDatabaseArchive = "BioModels_Database-r31_pub-sbml_files"
+						sh "aws s3 --no-progress cp --recursive ${env.S3_RELEASE_DIRECTORY_URL}/supplementary_files/${biomodelsDatabaseArchive}.tar.bz2 ."
+						// Runs perl script and then moves the models2pathways.tsv file to the downloads directory.
 						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-							sh "perl biomodels.pl -db ${env.RELEASE_CURRENT}"
-							sh "cp models2pathways.tsv ${env.ABS_DOWNLOAD_PATH}/${currentRelease}/"
+							sh "perl biomodels.pl -db ${env.RELEASE_CURRENT_DB}"
```

Comment:
We should look into integrating the logic from this script into the Java BioModels release step.

Reply:
Agreed! I'll look into it to see if it looks complex - not sure why it hasn't been incorporated yet